### ISSUE TITLE
convert_calendar adds time coordinate to static variables

### ISSUE THREAD

**Issue by larsbuntemeyer** [pydata/xarray#7398]
### What is your issue?

I just recognized that `convert_calendar` adds a time coordinate to variables that have been static before, e.g.,

```python
import numpy as np
import pandas as pd
import xarray as xr

np.random.seed(0)
temperature = 15 + 8 * np.random.randn(2, 2, 3)
precipitation = 10 * np.random.rand(2, 2, 3)
orog = 0 + 100 * np.random.randn(2, 2)
lon = [[-99.83, -99.32], [-99.79, -99.23]]
lat = [[42.25, 42.21], [42.63, 42.59]]
time = pd.date_range("2014-09-06", periods=3)
reference_time = pd.Timestamp("2014-09-05")


ds = xr.Dataset(
    data_vars=dict(
        temperature=(["x", "y", "time"], temperature),
        precipitation=(["x", "y", "time"], precipitation),
        orog = (["x", "y"], orog)
    ),
    coords=dict(
        lon=(["x", "y"], lon),
        lat=(["x", "y"], lat),
        time=time,
        reference_time=reference_time,
    ),
    attrs=dict(description="Weather related data."),
)
ds
```
![grafik](https://user-images.githubusercontent.com/5659125/209124247-4d6bb64f-8396-4a90-bcce-c420b022b0cf.png)
I would like to convert the time axis to cftime objects using
```python
ds.convert_calendar(ds.time.dt.calendar, use_cftime=True)
ds
```
![grafik](https://user-images.githubusercontent.com/5659125/209124475-2ca60508-2d39-432b-816d-0f2718cda5b4.png)
However, this results in an unexpected time coordinate in the `orog` variable...

**Comment by larsbuntemeyer** [comment]
It seems like this happens when the `time` dimensions is broadcasted during masking:
https://github.com/pydata/xarray/blob/41fef6f1352be994cd90056d47440fe9aa4c068f/xarray/coding/calendar_ops.py#L207
Cosequently is does not happen when i have `align_on="year"`...

**Comment by kmuehlbauer** [comment]
@larsbuntemeyer It looks like this has been fixed somewhere on the way to xarray 2024.10.0. At least your MCVE works now. Could you please confirm? This would also affect your PR #7399.

**Comment by larsbuntemeyer** [comment]
@kmuehlbauer i can confirm that this has been fixed! Thanks!
